### PR TITLE
Allow uploading of already extracted models

### PR DESCRIPTION
**Pull request by msamogh**
**Proposed changes**:
- Allow passing the path of an extracted model to `PUT /model`

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/agent.py [hunk 1]
```diff
@@ -892,7 +887,7 @@ def load_local_model(
         if os.path.isfile(model_path):
             model_archive = model_path
         else:
-            model_archive = get_latest_model(model_path)
+            model_archive = get_model(model_path)
```
Comment (tmbo) [inline rasa/core/agent.py]: this change doesn't work like this, `get_model` unpacks the model, but it is unpacked again in https://github.com/RasaHQ/rasa/pull/4625/files#diff-96e44592018d9ea0a68a83fad0d220a5L902

Comment (msamogh) [reply rasa/core/agent.py]: Oops, yes, misread it to mean it's directly using the folder as the model.

File: rasa/core/agent.py [hunk 2]
```diff
@@ -271,6 +271,14 @@ def _load_and_set_updated_model(
         raise
 
 
+def _contains_unpacked_model(model_path: Text) -> bool:
+    try:
+        core_path, nlu_path = get_model_subdirectories(model_path)
```
Comment (ricwo) [inline rasa/core/agent.py]:
```suggestion
        _ = get_model_subdirectories(model_path)
```

File: tests/test_server.py [hunk 3]
```diff
@@ -750,6 +750,28 @@ def test_load_model(rasa_app: SanicTestClient, trained_core_model):
...
+def test_load_unpacked_model(rasa_app: SanicTestClient, trained_core_model):
+    _, response = rasa_app.get("/status")
+
+    assert response.status == 200
+    assert "fingerprint" in response.json
+
+    old_fingerprint = response.json["fingerprint"]
+
+    with unpack_model(trained_core_model) as unpacked_model:
+        data = {"model_file": unpacked_model}
```
Comment (ricwo) [inline tests/test_server.py]: I think we should support a new json parameter here, e.g. `model_folder`. we can leave the loading unchanged in `Agent` but get the path in the endpoint with something like

```python
rjs = request.json
model_path = rjs.get("model_file") or rjs.get("model_folder")
```

File: rasa/core/agent.py (hunk 2 above)
Comment (msamogh) [reply rasa/core/agent.py]: Good point haha, for some reason I was like I have to make it `_, _` and that'd be weird.

File: tests/test_server.py (hunk 3 above)
Comment (msamogh) [reply tests/test_server.py]: Do you have any thoughts on handling such parameter interdependencies in the HTTP API spec? I came across this unresolved discussion here https://github.com/OAI/OpenAPI-Specification/issues/256, but don't have any strong preferences of my own yet.

**Comment by tmbo** [comment]
@akelad this is not linked to an issue - do you know where this is coming from?

**Comment by akelad** [comment]
hm no, not sure

**Comment by tmbo** [comment]
I'll close this for now - I don't see how this is needed without an attached issue.

**Pull request closed without merging**